traffic_portal/app/src/common/modules/form/server/FormServerController.js

```diff
@@ -158,6 +161,40 @@ var FormServerController = function(server, $scope, $location, $state, $uibModal
 
     $scope.hasPropertyError = formUtils.hasPropertyError;
 
+    $scope.oneInputHasPropertyError = formUtils.oneInputHasPropertyError;
+
+    $scope.elementId = '';
+
+    $scope.toggleIpAddressRequired = function(elementId) {
+        var elementToChange = elementId === 'ipAddress' ? 'ip6Address' : 'ipAddress';
+
+        if (document.getElementById(elementId).value === '') {
+            document.getElementById(elementToChange).required = true;
+        } else {
+            document.getElementById(elementToChange).required = false;
+        }
+    };
+
+    $scope.toggleIpGatewayRequired = function(elementId) {
+        var elementToChange = elementId === 'ipGateway' ? 'ip6Gateway' : 'ipGateway';
+
+        if (document.getElementById(elementId).value === '') {
+            document.getElementById(elementToChange).required = true;
+        } else {
+            document.getElementById(elementToChange).required = false;
+        }
+    };
+
+    $scope.isIPv4 = function() {
```

Comment:
maybe you should rename these to hasIPv4 because this test is not really checking if it "IS" IPv4

Reply:
done